### PR TITLE
Implement date/time truncation functions

### PR DESCRIPTION
**Pull request by ttnghia**
This implements the functions to truncate date/timestamp to some specific component, matching the Spark SQL function `trunc` and `date_trunc`.

Due to changes to related code, the existing module `DateTimeRebase` is changed to `DateTimeUtils` (without breaking), which will also contain the newly implemented function.

### CODE REVIEW

**Comment by ttnghia** [comment]
build

File: src/main/java/com/nvidia/spark/rapids/jni/DateTimeUtils.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+   *  - {@code "MINUTE"}: zero out the second with fraction part.<br>
+   *  - {@code "SECOND"}: zero out the second fraction part.<br>
+   *  - {@code "MILLISECOND"}: zero out the microseconds.<br>
+   *  - {@code "MICROSECOND"}: keep everything.<br>
+   *
+   * @param datetime The input date/time
+   * @param format The time component to truncate to
+   * @return The truncated date/time
+   */
+  public static ColumnVector truncate(ColumnView datetime, ColumnView format) {
```
Comment (revans2) [inline src/main/java/com/nvidia/spark/rapids/jni/DateTimeUtils.java:89]: Could we have a version that supports a scalar format? I just think that will be the most common use case and if we can save memory and time by not exploding the scalar into a column I think that would be great.

I think that can be done as a follow on issue if you want.

**Review by revans2 — APPROVED** [review]
Just a possible improvement that is minor.

File: src/main/java/com/nvidia/spark/rapids/jni/DateTimeUtils.java (hunk 1 above)
Comment (ttnghia) [reply src/main/java/com/nvidia/spark/rapids/jni/DateTimeUtils.java:89]: I thought about that, however, either parameter can be Scalar thus if doing so we will have to implement 4 JNI overloads for this function. Instead, the code supports columns in which any of them can have one row and can achieve the same performance. 

In plugin code, I just "convert"  Scalar into a column of one row. Such conversion should not be very expensive. But I can add more overload to support Scalar if that can help anything better.

Comment (revans2) [reply src/main/java/com/nvidia/spark/rapids/jni/DateTimeUtils.java:89]: Yes either can be a scalar, but fmt is the one that is really common to be a scalar. Most of the time you want to truncate to a very specific value so you know how to deal with the resulting timestamp/date.

**Pull request merged**